src/components/browse/singleDatasetView/singleDatasetView.tsx

```diff
@@ -30,21 +30,21 @@ const SingleDatasetView = () => {
     const getHits = (accessionId: string | null | undefined) => {
       if (accessionId && accessionId !== null && !queried) {
         setQueried(true);
-        querySearchService(setSearchResults, [], accessionId, 0, 1, "Dataset")
+        querySearchService(setSearchResults, [{key: "accession", value: accessionId}], "*", 0, 1, "Dataset")
       }
     };
     const getDetails = (datasetId: string | undefined) => {
       if (datasetId && paramId) {
         getDatasetDetails(datasetId, true, setDetails);
       }
     };
-    const processHits = (internalId: searchResponseModel | null) => {
-      if (internalId && internalId !== null && internalId.count === 1) {
+    const processHits = (searchResults: searchResponseModel | null) => {
+      if (searchResults && searchResults !== null && searchResults.count >= 1) {
```

Comment:
I think this should be `searchResults.count === 1` now.

Reply:
Well, not sure. If there is any weird case that there are 2 datasets with the same accession ID it would break. Maybe there needs to be a second check to display an error if there are more than 2 results?

Reply:
My guess is there will always be just one dataset associated with one accession ID. This is accession ID and not EGA accession ID. We are generating an unique ID and using it as an alias for the uuid.